lincheck/src/main/java/org/jetbrains/kotlinx/lincheck/Actor.kt

```diff
@@ -36,8 +35,31 @@ data class Actor(
     val arguments: List<Any?>,
     val handledExceptions: List<Class<out Throwable>>
 ) {
+    val wasInvoked: Boolean
+        get() = wasInvokedInner
+
+    private var wasInvokedInner = false
```

Comment:
Actors should specify scenarios and are immutable by design. Please, do not break this.

Reply:
I agree, but it still is not easy not to break other invariants and restrictions.
A alternative solution may be to make TestThreadExecution know about Strategy, add onActorStart(iThread) method to Strategy (even to StressStrategy) and call it before every actor.

Reply:
You can add `onActorStart(..)` method invocations using the byte-code generator. In this case, the `Strategy` class should have this method (`TestThreadExecution` can know about `Strategy` but not about specific implementations).

Reply:
Would such virtual calls be optimized out by the optimizer for StressStrategy?